src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -212,9 +212,15 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             genCodeForDivMod(treeNode->AsOp());
             break;
 
+        case GT_AND:
+#if !defined(TARGET_64BIT)
+            // An And that is not contained should not have any contained children.
+            assert(!treeNode->AsOp()->gtOp1->isContained() && !treeNode->AsOp()->gtOp2->isContained());
+            FALLTHROUGH;
```

Comment:
This is failing on arm32, but I see no reason to have this assert/these changes.